src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -498,11 +498,8 @@ else if (!instanceName.equals(initInstanceName)) {
                     isUsedVariableDeclarationFound = variableName.equals(currentVariableName);
                     break;
 
-                case TokenTypes.SEMI:
-                    break;
-
                 default:
-                    result = false;
+                    result = currentSiblingAst.getType() == TokenTypes.SEMI;
```

Comment:
This refactoring makes code less readable, creates questions when you read it and error-prone to extend the logic in the main loop if there is such a need. I'd vote for not following pitest here.

@romani thoughts?

Reply:
> error-prone to extend the logic in the main loop if there is such a need.

Extending any logic in this check is quite difficult. This loop is just for detecting if the statements result in an initialization sequence or not. Have you seen https://github.com/checkstyle/checkstyle/pull/13120#issuecomment-1579916446?

We can either not follow pitest here or go with the current solution. In any case, I don't think we should get rid of `result` as a loop-controlling variable.

Reply:
> Have you seen https://github.com/checkstyle/checkstyle/pull/13120#issuecomment-1579916446?

Yes, would you remember this when you open this code in few months?

Among 3 options we have - not follow pitest, remove `result`, move case logic to `default` - I wouldn't follow pitest here, you sacrifice readability and simplicity over making a tool happy.

Reply:
I am ok with not following pitest here:
1. It decreases performance by removing `result` (only on paper).
2. It reduces readability.

Reply:
should we go with removing the `result`?

Reply:
Let me review this.
I need to meditate a bit on this.

Reply:
@romani ping

Reply:
after meditation on this.
General code of this method is not good, tools by all means point us to this code to rewrite it.
We do not have time to rewrite code to better (it will be even outside of this method).
If rewriting of this code is inevitable, lets not have pitest suppressions as area for next engineer to extend it more.
So in next iteration over such code we will be constrained by tools from the beginning and it will force us to write better logic.

Reply:
So, we make bad code to be worse to enforce us to make it better when we rewrite it :) The problem with this approach is that most of the code will never be rewritten and making it worse by following tools will eventually lead to codebase which is hard to read, understand and maintain.

I'm ok to merge this specific case, but there should be some boundaries.

Reply:
@Kevin222004 why did you rewrite the whole logic to if conditions? We agreed to go with `result = currentSiblingAst.getType() == TokenTypes.SEMI;` in the default statement.

Reply:
@rdiachenko due to ci failure.
I am sorry, I am not remembering an exact message.